### PR TITLE
Forms:changes

### PR DESCRIPTION
**Pull request by Paul-tes**
- Createed a method called current_user in your ApplicationController to make current user data available to all controllers.
   - It will return the first user from the database.
- Created forms to perform the following functions:
   - Created a Post on behalf of the current_user (use the method that you created in your ApplicationController).
   - Created a comment on behalf of the current_user (use the method that you created in your ApplicationController).
- Added funtionality to Allow Users to add likes to Posts.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,10 +1,30 @@
...
+  end
+
+  def create
+    @post = Post.new(posts_params)
+    @post.author = current_user
+    if @post.save
+      redirect_to user_posts_path(id: @post.author_id)
+    else
+      render :new, alert: 'ERROR! something went wrong while creating the post'
+    end
```
Comment (levy002) [inline app/controllers/posts_controller.rb]: - [x] When creating a new post, it's crucial to associate the post with the correct user. In this case, we fetch the user using the `user_id` parameter provided. This step ensures that the post is linked accurately with the user who is currently creating it.

However, after creating the post, there might be an issue with redirection. The code redirects to the list of posts associated with the user who initiated the creation. However, due to potential discrepancies, such as using the `current_user` method, the redirection might lead to errors. This happens when the current user and the user associated with the newly created post don't match.

To address this, we explicitly fetch the correct user using the `User.find(params[:user_id])` approach. This way, we ensure that the redirection accurately displays the list of posts created by the user who just made the new post.

- ### ERROR AFTER CREATING POST WITH ANOTHER DIFFERENT TO THE `current_user`

image


```suggestion
   def create
    @post = Post.new(posts_params)
    @post.author = User.find(params[:user_id])
    if @post.save
      redirect_to user_posts_path(id: @post.author_id)
    else
      render :new, alert: 'ERROR! something went wrong while creating the post'
    end
```

**Review by levy002 — CHANGES_REQUESTED** [review]
## Hello @Paul-tes   , :wave:
### I'm Levy, your PR reviewer

Good job so far :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- No linter error :heavy_check_mark: 

## Status: REQUIRED CHANGES ♻️
_Check the comments under the review._


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @levy002 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -10,7 +10,7 @@ def new
 
   def create
     @post = Post.new(posts_params)
-    @post.author = current_user
+    @post.author = User.find(params[:user_id])
     if @post.save
       redirect_to user_posts_path(id: @post.author_id)
```
Comment (Gambit142) [inline app/controllers/posts_controller.rb:15]: - [OPTIONAL] Great work implementing the requests from the previous reviewer. I would advise you to revert to using the `current_user` variable that you were instructed to create in the `ApplicationController`. 👍 

---

The idea of the blog application is that users can create a post for themselves. A user should not be able to create a post on behalf of others as you see on Twitter, Facebook and the rest. In an ideal situation, a user will **_signup/login_** to the application and create posts that will be linked to that particular user. The reason why the `current_user` was created was for you to simulate yourself as the `current_user` and create posts on behalf of the `current_user` imagining that you are the `current_user`. 😎 

---

As for the error you were getting, you need the use the current_user.id in the URL helper that you are calling after a post is saved to the database 💻:

```ruby
def create
  @post = Post.new(posts_params)
  @post.author = current_user
  if @post.save
    redirect_to user_posts_path(id: current_user.id)
  else
    render :new, alert: 'ERROR! something went wrong while creating the post'
  end
end
```

**Review by Gambit142 — APPROVED** [review]
Hi @Paul-tes 👋,

## Status: Approved 🟢   

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

### Highlights
- Setup linters correctly ✔️ 
- Created form for Post model ✔️ 
- Created form for Comment model ✔️ 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 2 above)
Comment (Paul-tes) [reply app/controllers/posts_controller.rb:15]: I apreciate thanks bro.

**Pull request merged**